### PR TITLE
[Hub Generated] Review request for Microsoft.ContainerService to add version stable/2019-08-01

### PR DESCRIPTION
**Pull request by xuto2**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/xuto2/azure-rest-api-specs/dev-containerservice-Microsoft.ContainerService-2019-08-01...master/).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

**NOTE for reviewer**    
The net change is in this commit: https://github.com/Azure/azure-rest-api-specs/pull/6859/commits/76e439861b2089695228433d487b8a27ff8e937c

The rest are copied from the previous version (2019-06-01) as base.

### CODE REVIEW

File: specification/containerservice/resource-manager/readme.md [hunk 1]
```diff
@@ -34,15 +34,26 @@ These are the global settings for the ContainerServices API.
...
+  - Microsoft.ContainerService/stable/2017-07-01/containerService.json
+  - Microsoft.ContainerService/stable/2019-08-01/location.json
+  - Microsoft.ContainerService/stable/2019-08-01/managedClusters.json
+```
 ### Tag: package-2019-06
 
 These settings apply only when `--tag=package-2019-06` is specified on the command line.
 
-```yaml $(tag) == 'package-2019-06'
+``` yaml $(tag) == 'package-2019-06'
```
Comment (mboersma) [inline specification/containerservice/resource-manager/readme.md]: I assume everything still parses ok with this added space?

**Review by mboersma — APPROVED** [review]
lgtm

File: specification/containerservice/resource-manager/readme.md (hunk 1 above)
Comment (xuto2) [reply specification/containerservice/resource-manager/readme.md]: Yes, I can see the parsed file:
https://github.com/Azure/azure-rest-api-specs/blob/2307de018692dde013d3bf3ed0557c6d8f84519d/specification/containerservice/resource-manager/readme.md

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json [hunk 2]
```diff
@@ -0,0 +1,2349 @@
...
+      },
+      "description": "Properties of the managed cluster."
+    },
+    "ManagedClusterPrivateLinkResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The fully qualified Azure resource ID."
+        },
+        "name": {
```
Comment (ravbhatnagar) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: name can be obtained from the id. why do you need user to pass in name, type etc. when all these can be obtained from the id.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json [hunk 3]
```diff
@@ -0,0 +1,2349 @@
...
+          "type": "string",
+          "description": "The private link resource name."
+        },
+        "type": {
+          "type": "string",
+          "description": "The private link resource type."
+        },
+        "groupId": {
```
Comment (ravbhatnagar) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: how is groupId different from the name?

**Review by ravbhatnagar — COMMENTED** [review]
@xuto2 couple of comments. please take a look.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json (hunk 2 above)
Comment (xuto2) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: this is not ask for user input but allows AKS RP to pass back the info to customer(NRP in this particular case)

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json (hunk 3 above)
Comment (xuto2) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: this private link resource contract id dictated by NRP as this is a special scenario that requires joint effort from NRP/AKS RP.

I'm not expert on this feature - my understanding is that this group is a concept required by NRP to manage the private link resources.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json (hunk 2 above)
Comment (ravbhatnagar) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: are these read-only properties then? if so, can you please mark them as such

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json (hunk 3 above)
Comment (ravbhatnagar) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: is this a settable property?

Comment (ravbhatnagar) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: how does the customer know what goes here?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json [hunk 4]
```diff
@@ -1762,16 +1824,68 @@
...
```
Comment (ravbhatnagar) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: schema of private link resource must comply with that of ARM resource. id, name, type top level properties and then all other properties must be under "properties" property. Is the privateLinkResource being created as part of creation of managedCluster? I am assuming not. And so, if this is just a reference to an existing privateLinkResource, why is the reference not done via just the resourceid. Is there an RBAC element to this as well? The way you have modeled it, someone who does not have read access to the private endpoint resource will be able to read all information about it. 
Apologies for my limited understanding of how private link resource in NRP works.  I am assuming there must be a standard contract between NRP and every RP that wants to leverage private links.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json [hunk 5]
```diff
@@ -1762,16 +1824,68 @@
...
+          "description": "Properties related to private link cluster."
+        },
+        "apiServerAccessProfile": {
+          "$ref": "#/definitions/ManagedClusterAPIServerAccessProfile",
+          "description": "Access profile for managed cluster API server."
         }
       },
       "description": "Properties of the managed cluster."
     },
+    "ManagedClusterPrivateLinkResource": {
```
Comment (ravbhatnagar) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: Is there a benefit of modeling this as a child resource type under managedClusters and have GET APIs. Similar to how SQL is doing it - https://github.com/Azure/azure-rest-api-specs/pull/6866/

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json (hunk 3 above)
Comment (xuto2) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: this is set by AKS RP service. Customer just read the value.

Is there any impact to ARM?

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json (hunk 2 above)
Comment (xuto2) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: @weinong  your input is needed here

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json (hunk 4 above)
Comment (xuto2) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: @weinong  your input is needed here

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json (hunk 5 above)
Comment (xuto2) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: @weinong  can you comment here?

**Comment by weinong** [comment]
@xuto2 let's remove the private link resource. We will add it as a sub resource like SQL does in next iteration.
@ravbhatnagar we will remove private link resources

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json (hunk 2 above)
Comment (xuto2) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: we decided to remove this in this version.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json (hunk 4 above)
Comment (xuto2) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: we decided to remove this in this version.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json (hunk 5 above)
Comment (xuto2) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: we decided to remove this in this version.

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json (hunk 3 above)
Comment (xuto2) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json]: we decided to remove this section in this api version.

**Comment by ravbhatnagar** [comment]
Great, signing off from ARM side.

**Comment by xuto2** [comment]
@sergey-shandar Can you take a look? The net change is just one commit: https://github.com/Azure/azure-rest-api-specs/commit/76e439861b2089695228433d487b8a27ff8e937c

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**